### ISSUE TITLE
Reloading error sometimes

### ISSUE THREAD

**Issue by drager** [preactjs/preact-cli#1710]
I'm not sure if this is a bug with hot reloading or not, but sometimes I'm getting the following error (happens a lot when I make some larger code changes).

```
 Build  [=============       ] 65% (19.3s) building<i> [webpack-dev-middleware] wait until bundle finished: /backtester
 Build  [==================  ] 90% (37.8s) additional chunk assets processing<e> [webpack-dev-middleware] TypeError: Cannot read property '_value' of undefined
<e>     at module.exports (/home/drager/development/tinker/tinker-web-app/node_modules/preact-cli/lib/lib/webpack/create-load-manifest.js:10:52)
<e>     at /home/drager/development/tinker/tinker-web-app/node_modules/preact-cli/lib/lib/webpack/push-manifest.js:12:22
<e>     at _next1 (eval at create (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
<e>     at eval (eval at create (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:51:1)
<e>     at emit (/home/drager/development/tinker/tinker-web-app/node_modules/fork-ts-checker-webpack-plugin/lib/index.js:274:17)
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:47:1)
<e>     at Compiler.emitAssets (/home/drager/development/tinker/tinker-web-app/node_modules/webpack/lib/Compiler.js:491:19)
<e>     at onCompiled (/home/drager/development/tinker/tinker-web-app/node_modules/webpack/lib/Watching.js:51:19)
<e>     at /home/drager/development/tinkertinker-web-app/node_modules/webpack/lib/Compiler.js:681:15
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
<e>     at /home/drager/development/tinker/tinker-web-app/node_modules/webpack/lib/Compiler.js:678:31
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
<e>     at AsyncSeriesHook.lazyCompileHook (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/Hook.js:154:20)
<e>     at /home/drager/development/tinker/tinker-web-app/node_modules/webpack/lib/Compilation.js:1423:35
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
<e>     at AsyncSeriesHook.lazyCompileHook (/home/drager/development/tinker/tinker-web-app/node_modules/tapable/lib/Hook.js:154:20)
```

```
Environment Info:
  System:
    OS: Linux 5.3 Ubuntu 18.04.6 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz
  Binaries:
    Node: 16.4.0 - ~/.nvm/versions/node/v16.4.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.4.0/bin/yarn
    npm: 7.18.1 - ~/.nvm/versions/node/v16.4.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Firefox: 102.0
  npmPackages:
    preact: ^10.3.1 => 10.9.0 
    preact-cli: ^3.4.0 => 3.4.0 
    preact-render-to-string: ^5.1.4 => 5.2.0 
    preact-router: ^4.0.1 => 4.0.1 
```

**Comment by rschristian** [comment]
Hm, looks to be a race condition with the asset manifest being generated:

https://github.com/preactjs/preact-cli/blob/3f389c2486a3790ae9e18b67ac959f87de19c973/packages/cli/lib/lib/webpack/create-load-manifest.js#L10

Preloading is prod-only, so that plugin should probably only be returning an empty object in dev.

**Comment by drager** [comment]
> 

Hmm ok. I'm running via `preact watch --refresh --port 3000"`.

**Comment by rschristian** [comment]
Let me know if you can reliably reproduce it, though I somewhat doubt that's possible. I assume another save/HMR tick fixes this?

I think an early return is all that's needed there, will try to get something together tomorrow.

**Comment by drager** [comment]
> Let me know if you can reliably reproduce it, though I somewhat doubt that's possible. I assume another save/HMR tick fixes this?
> 
> I think an early return is all that's needed there, will try to get something together tomorrow.

Yeah, not sure how to cause it. Seems to happen more when I have code that contains errors... I actually need to restart the whole dev server to get it working again. It hangs for me...

I'll try using `patch-package`. Thanks!

**Comment by drager** [comment]
Used the patch for 1 hour now and didn't get any errors for this hour. Without the patch I got the error every now and then, so it seems to be working :tada:

**Comment by egilll** [comment]
I get this error every single time I introduce a TypeScript error (but only if Preact CLI has been running for a while; it does not appear when it's only been running for a few minutes).

My guess was therefore that this had something to do with the output from `fork-ts-checker-webpack-plugin` messing with the next plugin, even though that would be very strange behavior.

The error is not resolved by saving files again, instead I need to restart Preact CLI each time. In the meantime, Preact CLI will serve the old files, which is often quite confusing.
 
(I am running Preact Cli 3.4.0, TypeScript 3.9.9)

**Comment by Julioevm** [comment]
I'm having the same issue

**Comment by oliverfindl** [comment]
Hello,

I'm also having this same issue.
```
Environment Info:
  System:
    OS: Linux 4.15 Ubuntu 16.04.7 LTS (Xenial Xerus)
    CPU: (4) x64 Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz
  Binaries:
    Node: 17.9.0 - /usr/bin/node
    Yarn: 1.22.15 - /usr/bin/yarn
    npm: 8.8.0 - /usr/bin/npm
  Browsers:
    Chrome: 103.0.5060.134
    Firefox: 88.0
  npmPackages:
    preact: ^10.6.6 => 10.10.0
    preact-cli: ^3.4.0 => 3.4.0
    preact-render-to-string: ^5.1.20 => 5.2.1
    preact-router: ^4.0.1 => 4.1.0 
```

I can consistently reproduce it by re-declaring some type, e.g.:
```typescript
type Test1 = any;
type Test1 = any;
```
Which throws kinda standard type error and after I fix it, e.g.:
```typescript
type Test1 = any;
type Test2 = any;
```
Then Webpack hangs with error:
```
[webpack-dev-middleware] TypeError: Cannot read properties of undefined (reading '_value')
<e>     at module.exports (.../node_modules/preact-cli/lib/lib/webpack/create-load-manifest.js:10:52)
<e>     at .../node_modules/preact-cli/lib/lib/webpack/push-manifest.js:12:22
<e>     at _next1 (eval at create (.../node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
<e>     at eval (eval at create (.../node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:51:1)
<e>     at emit (.../node_modules/fork-ts-checker-webpack-plugin/lib/index.js:274:17)
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (.../node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:47:1)
<e>     at Compiler.emitAssets (.../node_modules/webpack/lib/Compiler.js:491:19)
<e>     at onCompiled (.../node_modules/webpack/lib/Watching.js:51:19)
<e>     at .../node_modules/webpack/lib/Compiler.js:681:15
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (.../node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
<e>     at .../node_modules/webpack/lib/Compiler.js:678:31
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (.../node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
<e>     at AsyncSeriesHook.lazyCompileHook (.../node_modules/tapable/lib/Hook.js:154:20)
<e>     at .../node_modules/webpack/lib/Compilation.js:1423:35
<e>     at AsyncSeriesHook.eval [as callAsync] (eval at create (.../node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
<e>     at AsyncSeriesHook.lazyCompileHook (.../node_modules/tapable/lib/Hook.js:154:20)
```

And won't reload itself after another change. It needs to be killed and started again.

[Fix](https://github.com/preactjs/preact-cli/pull/1711) mentioned above solves this issue for me.

Thanks.

**Comment by rschristian** [comment]
Fix released in v3.4.1.

Let me know if anyone still has any problems.
